src/main/java/hudson/plugins/matrix_configuration_parameter/shortcut/ResultShortcut.java

```diff
@@ -69,7 +70,7 @@ public ResultShortcut(String name, boolean exact, List<String> resultsToCheck) {
         this.name = Util.fixNull(name);
         this.exact = exact;
         this.resultsToCheck = (resultsToCheck != null)
-            ? Collections.unmodifiableList(resultsToCheck)
+            ? new ArrayList<>(resultsToCheck)
```

Comment:
Note for me:
Collections.unmodifiableList results following xml:
```
              <shortcutList class="java.util.Arrays$ArrayList">
                <a class="hudson.plugins.matrix_configuration_parameter.shortcut.MatrixCombinationsShortcut-array">
                  <hudson.plugins.matrix__configuration__parameter.shortcut.ResultShortcut>
                    <name>Successful</name>
                    <exact>false</exact>
                    <resultsToCheck class="java.util.Collections$UnmodifiableRandomAccessList" resolves-to="java.util.Collections$UnmodifiableList">
                      <c class="com.google.common.collect.Lists$TransformingRandomAccessList">
                        <fromList class="java.util.Arrays$ArrayList">
                          <a class="hudson.model.Result-array">
                            <hudson.model.Result>SUCCESS</hudson.model.Result>
                          </a>
                        </fromList>
                        <function class="hudson.plugins.matrix_configuration_parameter.shortcut.ResultShortcut$1"/>
                      </c>
                      <list class="com.google.common.collect.Lists$TransformingRandomAccessList" reference="../c"/>
                    </resultsToCheck>
                  </hudson.plugins.matrix__configuration__parameter.shortcut.ResultShortcut>
                  <hudson.plugins.matrix__configuration__parameter.shortcut.ResultShortcut>
                    <name>Failed</name>
                    <exact>false</exact>
                    <resultsToCheck class="java.util.Collections$UnmodifiableRandomAccessList" resolves-to="java.util.Collections$UnmodifiableList">
                      <c class="com.google.common.collect.Lists$TransformingRandomAccessList">
                        <fromList class="java.util.Arrays$ArrayList">
                          <a class="hudson.model.Result-array">
                            <hudson.model.Result>FAILURE</hudson.model.Result>
                          </a>
                        </fromList>
                        <function class="hudson.plugins.matrix_configuration_parameter.shortcut.ResultShortcut$1"/>
                      </c>
                      <list class="com.google.common.collect.Lists$TransformingRandomAccessList" reference="../c"/>
                    </resultsToCheck>
                  </hudson.plugins.matrix__configuration__parameter.shortcut.ResultShortcut>
                  <hudson.plugins.matrix__configuration__parameter.shortcut.MatrixCombinationsShortcut_-All/>
                  <hudson.plugins.matrix__configuration__parameter.shortcut.MatrixCombinationsShortcut_-None/>
                </a>
              </shortcutList>
```

This means `unmodifiableList` preserves the original structure (`TransformingRandomAccessList`).
`ArrayList` receives iterator and create a new array, and remove the original structure.

Reply:
Now visible in the `hairyCompat` test input.